mountpoint-s3/src/fs.rs

```diff
@@ -1168,8 +1168,8 @@ where
             }
             FileHandleState::Read { .. } => {
                 // TODO make sure we cancel the inflight PrefetchingGetRequest. is just dropping enough?
-                file_handle.inode.finish_reading()?;
                 metrics::decrement_gauge!("fs.current_handles", 1.0, "type" => "read");
+                file_handle.inode.finish_reading()?;
```

Comment:
Swapped these two lines since we have a small concern that in the future, `finish_reading` could fail and we wouldn't decrement the gauge. It currently doesn't do anything though.